core/domain/stats_domain.py

```diff
@@ -16,213 +16,250 @@
 
-            # check if interaction_id is valid
+            # Verify interaction_id is valid.
             if (self.interaction_id not in
                     interaction_registry.Registry.get_all_interaction_ids()):
                 raise utils.ValidationError(
-                    'Unknown interaction id %s' % self.interaction_id)
+                    'Unknown interaction_id: %s' % self.interaction_id)
+
+        if not isinstance(self.submitted_answer_list, list):
+            raise utils.ValidationError(
+                'Expected submitted_answer_list to be a list, received %s' %
+                str(self.submitted_answer_list))
+
+        if not isinstance(self.schema_version, int):
+            raise utils.ValidationError(
+                'Expected schema_version to be an integer, received %s' % str(
+                    self.schema_version))
+
+        if self.schema_version < 1:
+            raise utils.ValidationError(
+                'schema_version < 1: %d' % self.schema_version)
+
+        if self.schema_version > feconf.CURRENT_STATE_ANSWERS_SCHEMA_VERSION:
+            raise utils.ValidationError(
+                'schema_version > feconf.CURRENT_STATE_ANSWERS_SCHEMA_VERSION '
+                '(%d): %d' % (
+                    feconf.CURRENT_STATE_ANSWERS_SCHEMA_VERSION,
+                    self.schema_version))
+
+        for submitted_answer in self.submitted_answer_list:
+            submitted_answer.validate()
+
+
+class SubmittedAnswer(object):
+    """Domain object representing an answer submitted to a state."""
 
-        if not isinstance(self.answers_list, list):
+    # NOTE TO DEVELOPERS: do not use the rule_spec_str and answer_str
+    # parameters, as they are here as part of the answer migration.
+    def __init__(self, normalized_answer, interaction_id, answer_group_index,
+                 rule_spec_index, classification_categorization, params,
+                 session_id, time_spent_in_sec, rule_spec_str=None,
+                 answer_str=None):
+        self.normalized_answer = normalized_answer
+        self.interaction_id = interaction_id
+        self.answer_group_index = answer_group_index
+        self.rule_spec_index = rule_spec_index
+        self.classification_categorization = classification_categorization
+        self.params = params
+        self.session_id = session_id
+        self.time_spent_in_sec = time_spent_in_sec
+        self.rule_spec_str = rule_spec_str
+        self.answer_str = answer_str
+
+    @property
+    def json_size(self):
+        """Returns the size of the dict returned by to_dict(). This may not be
+        exactly the size of the returned dict, since it assumes the property in
+        the dict for the json_size is sys.maxint.
+        """
+        return sys.getsizeof(json.dumps(self.to_dict(json_size=sys.maxint)))
+
+    def to_dict(self, json_size=None):
+        submitted_answer_dict = {
+            'answer': self.normalized_answer,
```

Comment:
It was to avoid confusion with ``SubmittedAnswer`` containing an answer, but I agree with the consistency bit. For simplicity, I renamed ``normalized_answer`` to ``answer``.